src/pages/drivers/[driver].tsx

```diff
@@ -98,16 +122,51 @@ export default function Driver() {
         Error! :((((
       </p>
     )
+    //------------------
+    var wins = 0
+    var poles = 0
+    var points = 0
+    var FastestLaps = 0
+
+    for (let i = 0; i < resultsQuery.data.length; i++)
+    {
+      if (resultsQuery.data[i].Results[0].position === "1"){wins += 1}
+      if (resultsQuery.data[i].Results[0].grid === "1"){poles += 1}
+      // if (resultsQuery.data[i].Results[0].status !== "Withdrew")
+      // {
+      //   if (resultsQuery.data[i].Results[0].FastestLap.rank === "1"){FastestLaps += 1}
+      // }
```

Comment:
I would probably move this for loop out of the component and create a custom function. You're accessing similar things from the API, so you could just create one function that returns whatever you pass to it. 

I'd first create a schema based on the data we want to get: 

```ts
const resultsSchema = z.object({
  Constructor: z.object({
    constructorId: z.string(),
    name: z.string(),
    nationality: z.string(),
  }),
  Driver: z.object({
    code: z.string(),
    dateOfBirth: z.string(),
    driverId: z.string(),
    familyName: z.string(),
    givenName: z.string(),
    nationality: z.string(),
    permanentNumber: z.string(),
  }),
  grid: z.string(),
  laps: z.string(),
  number: z.string(),
  points: z.string(),
  position: z.string(),
  positionText: z.string(),
  status: z.string(),
})
```

Then I'd narrow in on that and create a races schema to get the specific data we're interested in: 
```ts
const racesSchema = z.object({
  Circuit: z.object({
    Location: z.object({
      country: z.string(),
      lat: z.string(),
      locality: z.string(),
      long: z.string(),
    }),
  }),
  Results: z.array(resultsSchema),
  date: z.string(),
  raceName: z.string(),
  round: z.string(),
  season: z.string(),
})
```
and create a TypeScript type from it: 

```ts
type Races = z.infer<typeof racesSchema>
```

I'd then create a helper function that essentially just adds up the amount of things you want: 

```ts
type Kind = 'position' | 'grid'
function getDriverStats(
  results: Races[],
  key: Kind,
  value: number | string
) {
  return results?.reduce((total, current) => {
    if (current?.Results[0][key] === value) {
      total += 1
    }
    return total
  }, 0)
}
```

Then, in the Driver component, you can access these values like so: 
```tsx
const wins = getDriverStats(
  resultsQuery.data,
  'position',
  '1'
)
const poles = getDriverStats(
  resultsQuery.data,
  'grid',
  '1'
)
```

Reply:
Done